tests/test_main.py

```diff
@@ -206,7 +216,17 @@ def run(self, parsed_args):
                 """
             ),
         ],
-        ["type: charm", None],
+        [
+            dedent(
+                """\
+                type: charm
+                name: test-charm-name-from-metadata-yaml
```

Comment:
```suggestion
                name: test-charm-name-from-charmcraft-yaml
```

if I'm reading this correctly?

Reply:
Yes